### PR TITLE
Check if resolver needs imports and if bridge needs sync separately

### PR DESCRIPTION
**Pull request by ajsantander**
The connect-bridge was not syncing cache on the bridge if its resolver didnt need to import any addresses. However, if for example the bridge has a new dependency, its cache might be out of sync. So the script now separately checks if the bridge's cache is out of sync.

### CODE REVIEW

**Review by i-stam — APPROVED** [review]
lgtm, perhaps we could somehow test it to be sure that it works as expected

File: publish/src/commands/connect-bridge.js [hunk 1]
```diff
@@ -127,41 +127,68 @@ const connectLayer = async ({
...
+	}
+
+	// ---------------------------------
+	// Sync cache on bridge if needed
+	// ---------------------------------
+
+	let needToSyncCacheOnBridge = needToImportAddresses;
+	if (!needToSyncCacheOnBridge) {
+		const isResolverCached = await SynthetixBridge.methods
+			.isResolverCached(AddressResolver.options.address)
```
Comment (justinjmoses) [inline publish/src/commands/connect-bridge.js:171]: This has changed now due to SIP-100

File: publish/src/commands/connect-bridge.js [hunk 2]
```diff
@@ -127,41 +127,68 @@ const connectLayer = async ({
...
+		}
+	}
+
+	if (needToSyncCacheOnBridge) {
+		console.log(yellow('  > Syncing cache on bridge...'));
+
+		if (!dryRun) {
+			console.log(yellow('  > SynthetixBridge.setResolverAndSyncCache()...'));
+			tx = await SynthetixBridge.methods
+				.setResolverAndSyncCache(AddressResolver.options.address)
```
Comment (justinjmoses) [inline publish/src/commands/connect-bridge.js:184]: (ibid - from SIP-100)

**Review by justinjmoses — CHANGES_REQUESTED** [review]

File: publish/src/commands/connect-bridge.js (hunk 1 above)
Comment (ajsantander) [reply publish/src/commands/connect-bridge.js:171]: Updated

File: publish/src/commands/connect-bridge.js (hunk 2 above)
Comment (ajsantander) [reply publish/src/commands/connect-bridge.js:184]: Updated

**Review by justinjmoses — APPROVED** [review]

**Pull request merged**